cpp/oneapi/dal/backend/primitives/objective_function/test/fixture.hpp

```diff
@@ -291,6 +291,18 @@ class logloss_test : public te::float_algo_fixture<Param> {
         }
     }
 
+    float_t clip_prob(float_t prob) {
```

Comment:
Please add the comment what this function does and for what.
Something like "Fit the input probability value into the interval [epsilon, 1.0 - epsilon]. This is needed because too small or too close to 1.0 probability values can lead to infinity results in logloss computations leading to test failures".
And I agree to Andreas's comment: need a comment about the choice of 'epsilon' values. Even if they were taken from sklearn, ok. Please write it in the comment.
Also it would be good to give them names, i.e. assign to some constexpr variables. Having magic numbers in the code is not recommended by our coding guidelines.

Reply:
Sorry, should have doubled down on this. We need to explanation in code, not just in GH :) Agree with @Vika-F that @avolkov-intel should add this in a follow-up PR. Thanks!